src/main/sensors/esc_sensor.c

```diff
@@ -3330,6 +3337,136 @@ static void recordSensorProcess(timeUs_t currentTimeUs)
 }
 
 
+/*
+ * Castle Live Link Telemetry Data Recorder
+ */
+
+#define CASTLE_BATTERY_VOLTAGE_SCALE 20000.0f // in mV, not volts
+#define CASTLE_RIPPLE_VOLTAGE_SCALE 4000.0f  // in mV, not volts
+#define CASTLE_BATTERY_CURRENT_SCALE 50000.0f // in mA, not amps
+#define CASTLE_THROTTLE_SCALE 1.0f // in ms.
+// Note the doc for POWER_SCALE says the unit is percent for scale 0.2502,
+// but that actually puts it in a 0..1 range, not 0..100.
+#define CASTLE_POWER_SCALE 250.2f // in per-thousands (mils)
+#define CASTLE_RPM_SCALE 20416.7f
+#define CASTLE_BEC_VOLTAGE_SCALE 4000.0f  // in mV, not volts
+#define CASTLE_BEC_CURRENT_SCALE 4000.0f  // in mA, not amps
+#define CASTLE_LIN_TEMP_SCALE 30.0f // in degrees C
+#define CASTLE_NTC_TEMP_SCALE 63.8125f // needs further processing.
+#define CASTLE_NTC_R0 10000.0f
+#define CASTLE_NTC_R2 10200.0f
+#define CASTLE_NTC_B 3455.0f
+
+// Castle Link telemetry is not digital; it's the length of time between the rising edge of the
+// output (inverted) pwm and the falling edge of the input "tick" generated by the ESC.  To
+// calibrate this time, each frame includes minimum-length (0.5ms) and 1ms calibration items.
+// The "max" below is required because a data item could be smaller than the 0.5ms calibration
+// item due to measurement error.
+#define CASTLE_DECODE_1(tele, item, scale, cal0p5) \
+    ((MAX(tele->item, cal0p5) - cal0p5) * scale / tele->oneMs)
+
+#define CASTLE_DECODE(tele, item, scale, cal0p5) \
+    CASTLE_DECODE_1(tele, item, CASTLE_##scale##_SCALE, cal0p5)
+
+#ifdef USE_TELEMETRY_CASTLE
+static float castleDecodeTemperature(castleTelemetry_t* tele)
+{
+    if (tele->linTempOrHalfMs < tele->ntcTempOrHalfMs) {
+        float value = CASTLE_DECODE(tele, ntcTempOrHalfMs, NTC_TEMP, tele->linTempOrHalfMs);
+        value = constrainf(value, 1, 255);
+        // This is from the castle link documentation.  It's the "beta" method of calculating NTC
+        // temperatures.
+        // Reference temp = 298K (Has to be really 298.15K = 25C)
+        // R0 = NTC resistance at reference temp = 10000
+        // R2 = Reference resistance for voltage divider ADC input = 10200
+        // (Why would they use a 10.2K and not a 10K resistor?)
+        // B = NTC Beta (Probably the B_25/100 value)
+        // 273 = 0C in K (Almost certainly really 273.15)
+        return (1 / (logf(value * CASTLE_NTC_R2 / (255 - value) / CASTLE_NTC_R0) / CASTLE_NTC_B + 1.0f / 298.0f)) - 273;
```

Comment:
Is the NTC calc function not suitable?

Reply:
Had to do some math but it does work.  I chose to use a delta based on 298K rather than exactly match Castle's calculation that's 0.15 kelvins off, since it makes little difference and avoids an arbitrary "- 0.15" in the code.  Also their calculation is probably just truncated for convenience anyway.